tests/python/pants_test/task/test_task.py

```diff
@@ -95,15 +105,17 @@ def test_incremental(self):
     vtB = task.execute()
     self.assertContent(vtB, one + two)
 
-    # Incremental atop existing directory for vtA.
+    # vtC.previous_cache_key == vtB.cache_key so the task appends the file to vtB's results.
+    # This behavior used to be skipped because the current_dir existed when checked by cache_manager._use_previous_dir.
```

Comment:
I thought about this a bit more... the idea was that if we switched back to exactly the same inputs as some previous build with zinc (for example), then zinc would quickly do the right thing: which is noop.

On the other hand, clearing the directory regardless as you're doing now is definitely less error prone then ever reusing an existing directory, so I think your change makes sense.

I would still love to see the test updated before landing this though.